client/question/components/SpeechBalloonDirectives.js

```diff
@@ -16,29 +16,32 @@
  * @fileoverview Directives for displaying speech balloons.
  */
 
-tie.directive('tieSpeechBalloonContainer', ['$timeout', function($timeout) {
-  return {
-    restrict: 'E',
-    scope: {},
-    link: function(scope, element) {
-      var speechBalloonContainerElement = element[0];
-      var speechBalloonContainer =
-          angular.element(speechBalloonContainerElement);
-      speechBalloonContainer.addClass('tie-speech-balloon-container');
-      speechBalloonContainerElement.style.opacity = '0';
-      speechBalloonContainerElement.style.transition = 'unset';
-      speechBalloonContainerElement.style.display = 'none';
-      $timeout(function() {
-        speechBalloonContainerElement.style.display = 'block';
-        speechBalloonContainerElement.style.marginTop =
-            '-' + speechBalloonContainerElement.offsetHeight.toString() + 'px';
+tie.directive('tieSpeechBalloonContainer', ['$timeout', 'DELAY_STYLE_REMOVAL',
+  function($timeout, DELAY_STYLE_REMOVAL) {
+    return {
+      restrict: 'E',
+      scope: {},
+      link: function(scope, element) {
+        var speechBalloonContainerElement = element[0];
+        var speechBalloonContainer =
+            angular.element(speechBalloonContainerElement);
+        speechBalloonContainer.addClass('tie-speech-balloon-container');
+        speechBalloonContainerElement.style.opacity = '0';
+        speechBalloonContainerElement.style.transition = 'unset';
+        speechBalloonContainerElement.style.display = 'none';
         $timeout(function() {
-          speechBalloonContainerElement.removeAttribute('style');
+          speechBalloonContainerElement.style.display = 'block';
+          speechBalloonContainerElement.style.marginTop =
+              '-' + speechBalloonContainerElement.offsetHeight.toString() +
+              'px';
+          $timeout(function() {
+            speechBalloonContainerElement.removeAttribute('style');
+          }, DELAY_STYLE_REMOVAL);
```

Comment:
Relevant changes are use of constants in line 19 and 39 (the lines in between are just spacing adjustments to pass lint test).